### ISSUE TITLE
arg `split` not defined in GeoDataset but required by GeoDataModule.setup

### ISSUE THREAD

**Issue by adriantre** [microsoft/torchgeo#1404]
### Description

There is lacking documentation for the `split` arg in `GeoDataset`. `GeoDataModule.setup` passes this arg to the dataset which will raise.

Either 
1. the datamodule should allow non-splitted dataset
2. dataset should take this arg and ignore it by default

Thoughts?

```python
Traceback (most recent call last):
  File "/.../lib/python3.9/site-packages/IPython/core/interactiveshell.py", line 3508, in run_code
    exec(code_obj, self.user_global_ns, self.user_ns)
  File "<ipython-input-14-d9594c5841bd>", line 1, in <module>
    GeoDataModule(CustomGeoDataset).setup("fit")
  File "/.../lib/python3.9/site-packages/torchgeo/datamodules/geo.py", line 124, in setup
    self.train_dataset = self.dataset_class(  # type: ignore[call-arg]
TypeError: __init__() got an unexpected keyword argument 'split'
```

### Steps to reproduce

```python
from rasterio.crs import CRS

from torchgeo.datasets import (
    BoundingBox,
    GeoDataset,
)
from torchgeo.datamodules import GeoDataModule


class CustomGeoDataset(GeoDataset):
    def __init__(
        self,
        bounds: BoundingBox = BoundingBox(0, 1, 2, 3, 4, 5),
        crs: CRS = CRS.from_epsg(4087),
        res: float = 1,
    ) -> None:
        super().__init__()
        self.index.insert(0, tuple(bounds))
        self._crs = crs
        self.res = res

    def __getitem__(self, query: BoundingBox) -> dict[str, BoundingBox]:
        hits = self.index.intersection(tuple(query), objects=True)
        hit = next(iter(hits))
        bounds = BoundingBox(*hit.bounds)
        return {"index": bounds}
    
    
if __name__ == '__main__':
    GeoDataModule(CustomGeoDataset).setup("fit")
```

### Version

0.4.1

**Comment by isaaccorley** [comment]
I believe the recommended way is to create your own `CustomGeoDataModule` and override the `setup` method. Something like:

```python
class CustomGeoDataModule(GeoDataModule):
   def setup(stage=None):
      self.dataset = dataset_class(**self.kwargs)
      self.sampler = SomeGeoSampler(...)
```

We also have tools for splitting your geodataset into train/val/test if you're interested. You can see the [L8BiomeDataModule](https://github.com/microsoft/torchgeo/blob/main/torchgeo/datamodules/l8biome.py) as an example of this.

**Comment by adriantre** [comment]
Thanks for the tip, makes sense. I will make my GeoDataset accept `split`-arg which I will eventually use anyway. And when using the same Dataset with `stage='predict'` I will just ignore the arg. 
And the `random_bbox_assignment` looks interesting, couldn't find it in the docs 😄

**Comment by adamjstewart** [comment]
Yes, the default setup method for our base class assumes that the dataset has a `split` argument that takes at least 3 possible values: train, val, and test. This was done during the last big refactor because that was the most common setup method among our existing datamodules. For any datasets that differ from this, it's necessary to override the setup method. This isn't well documented at the moment, but I would like to have better documentation for how to create your own datamodules or trainers like we have for our datasets.

> And the `random_bbox_assignment` looks interesting, couldn't find it in the docs 😄

That's because it hasn't been released yet. It's currently only available in the main branch, but you can see the docs [here](https://torchgeo.readthedocs.io/en/latest/api/datasets.html#splitting-functions). You can install main or use a locally checked-out copy if you want to play with this feature. It will be included in our 0.5 release (expected in August).

**Comment by adamjstewart** [comment]
I think this issue was addressed but feel free to reopen if that isn't the case.
